app/grandchallenge/reader_studies/views.py

```diff
@@ -867,6 +932,107 @@ def ground_truth(self, request, pk=None, case_pk=None):
+    def reader_study(self):
+        reader_study_pk = self.request.query_params.get("reader_study")
+        if reader_study_pk:
+            return ReaderStudy.objects.get(pk=reader_study_pk)
+
+    def get_serializer_class(self):
+        if self.action in ["partial_update", "update", "create"]:
+            return DisplaySetPostSerializer
+        return DisplaySetSerializer
+
+    def partial_update(self, request, pk=None):
+        instance = self.get_object()
+        if request.data.get("value"):
+            # Get the provided civ from the current reader study's display sets.
+            civs = instance.reader_study.display_sets.values_list(
+                "values", flat=True
+            )
+            civ = ComponentInterfaceValue.objects.filter(id__in=civs).get(
+                id=request.data.get("value")
+            )
+
+            # Because the civ is added to a display set, it can no longer be
+            # a value for another display set.
+            civ.display_sets.clear()
+
+            # If there is already a value for the provided civ's interface in
+            # this display set, remove it. There should not be any cases where
+            # multiple civs are assigned for the same interface, but loop over
+            # the qs just in case.
+            assigned_civs = instance.values.filter(interface=civ.interface)
+            for assigned in assigned_civs:
+                ds = DisplaySet.objects.create(
+                    reader_study=instance.reader_study
+                )
+                ds.values.add(assigned)
+                instance.values.remove(assigned)
+
+            # Add the provided civ to the current display set
+            instance.values.add(civ)
+
+            # Adding the civ to this display set removes it from another.
+            # Find any display sets with no values and delete them.
+            DisplaySet.objects.filter(
+                reader_study=instance.reader_study, values=None
+            ).delete()
+
+        if request.data.get("values"):
+            serialized_data = ComponentInterfaceValuePostSerializer(
+                many=True, data=request.data.get("values")
+            )
+            if serialized_data.is_valid():
+                civs = serialized_data.create(serialized_data.validated_data)
+                instance.values.remove(
+                    *instance.values.filter(
+                        interface__in=[civ.interface for civ in civs]
+                    )
+                )
+                instance.values.add(*civs)
+        serializer = self.get_serializer(instance)
+        return Response(serializer.data)
+
+    def get_queryset(self):
+        queryset = DisplaySet.objects.all().select_related("reader_study")
+        unanswered_by_user = self.request.query_params.get(
+            "unanswered_by_user"
+        )
+        if unanswered_by_user == "True":
+            queryset = queryset.exclude(answers__creator=self.request.user)
```

Comment:
Does this cover the case of partial answers? Eg, an editor adds a new question after the user has gone through the study.

Reply:
It does not. For the implementation: can you think of an instance where you'd want to filter by unanswered but not by reader study? The query will be a lot more efficient if we can just use a single reader study.

Reply:
The use case I'm thinking of (which does happen now I think) is:

- Editor creates a reader study with x questions
- Reader answers y display sets, creating x * y answers 
- Editor realises they missed something, creates a new question
- Reader opens the study, for the first y display sets they answer just the 1 new question with their existing answers in play, for display set > y they answer all x + 1 questions as they go

All the same reader study.